lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -520,6 +521,46 @@ static CallExpr *createContainerKeyedByCall(ASTContext &C, DeclContext *DC,
                                   C.AllocateCopy(argLabels));
 }
 
+/// A special helper function that looks for a field in targetDecl that is a
+/// non-static var decl. As an additional sanity check, we ensure that if the
+/// lookup that we perform /could/ succeed multiple times, we assert. Since the
+/// only purpose of this is testing the assert, this is only enabled in asserts
+/// builds.
+static VarDecl *lookupNonStaticProperty(NominalTypeDecl *targetDecl,
```

Comment:
This is overkill in my opinion. Please just add an 'isStatic()' check below.

Reply:
I disagree very strongly here. The lack of such things in the general case led to the bug here.

Reply:
Also you assertion will fire on invalid code. We only perform re-declaration checking on the primary file, and even in primary files, a re-declaration will set the 'invalid' flag on the re-declaration but not physically remove it from its parent decl. So if you want this to be 100% robust, you should also filter out 'invalid' declarations, and if there is more than one lookup result, just return the first one since we will diagnose the error elsewhere.

However, I disagree that adding a new 50-line method is a 'general case'. It's only used in one place, in one file. Unless you plan on re-working all existing callers of lookupDirect(), please do not do this.